org.qgis.qgis.json

```diff
@@ -67,6 +67,318 @@
+                        },
+                        {
+                            "type": "file",
+                            "url": "https://github.com/substrait-io/substrait/archive/v0.27.0.tar.gz",
+                            "sha256": "4ed375f69d972a57fdc5ec406c17003a111831d8640d3f1733eccd4b3ff45628",
+                            "dest-filename": "substrait_v0.27.0.tar.gz"
+                        },
+                        {
+                            "type": "file",
+                            "url": "https://archive.apache.org/dist/thrift/0.16.0/thrift-0.16.0.tar.gz",
+                            "sha256": "f460b5c1ca30d8918ff95ea3eb6291b3951cf518553566088f3f2be8981f6209",
+                            "dest-filename": "thrift-0.16.0.tar.gz"
+                        },
+                        {
+                            "type": "file",
+                            "url": "https://github.com/openucx/ucx/archive/v1.12.1.tar.gz",
+                            "sha256": "9bef31aed0e28bf1973d28d74d9ac4f8926c43ca3b7010bd22a084e164e31b71",
+                            "dest-filename": "ucx-1.12.1.tar.gz"
+                        },
+                        {
+                            "type": "file",
+                            "url": "https://github.com/JuliaStrings/utf8proc/archive/v2.7.0.tar.gz",
+                            "sha256": "4bb121e297293c0fd55f08f83afab6d35d48f0af4ecc07523ad8ec99aa2b12a1",
+                            "mirror-urls": [
+                                "https://github.com/JuliaStrings/utf8proc/archive/refs/tags/v2.7.0.tar.gz"
+                            ],
+                            "dest-filename": "utf8proc-v2.7.0.tar.gz"
+                        },
+                        {
+                            "type": "file",
+                            "url": "https://github.com/xtensor-stack/xsimd/archive/9.0.1.tar.gz",
+                            "sha256": "b1bb5f92167fd3a4f25749db0be7e61ed37e0a5d943490f3accdcd2cd2918cc0",
+                            "mirror-urls": [
+                                "https://github.com/xtensor-stack/xsimd/archive/refs/tags/9.0.1.tar.gz"
+                            ],
+                            "dest-filename": "xsimd-9.0.1.tar.gz"
+                        },
+                        {
+                            "type": "file",
+                            "url": "https://zlib.net/fossils/zlib-1.2.13.tar.gz",
+                            "sha256": "b3a24de97a8fdbc835b9833169501030b8977031bcb54b3b3ac13740f846ab30",
+                            "dest-filename": "zlib-1.2.13.tar.gz"
+                        },
+                        {
+                            "type": "file",
+                            "url": "https://github.com/facebook/zstd/releases/download/v1.5.5/zstd-1.5.5.tar.gz",
+                            "sha256": "9c4396cc829cfae319a6e2615202e82aad41372073482fce286fac78646d3ee4",
+                            "dest-filename": "zstd-1.5.5.tar.gz"
```

Comment:
zstd and probably some other of these dependencies are already present, why are you adding them again?

Reply:
They would need to be either installed as a module so they would be visible in the system during compile or there would need to be a way to point the arrow build process to the archives so it can compile them... and they would need to be compatible.

Honestly being wasteful on space and compile time here seems to be worth it to me to avoid hassle of trying to deduplicate dependencies but maybe it's just because I'm bad at this.